### PR TITLE
Edit workflows

### PR DESCRIPTION
**Pull request by itsacoyote**
# Description

Remove link-checker from Checks workflow. Removes the need to include a secret environment variable and build in the checks workflow. SEO Nuxt module is already configured to run a link checker on build.

Add environment tag to workflow for staging to track deployments.

Add codeowners to project. Requires DevXP team as PR reviewers on changes to project except for changes to /content and /public

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,9 +1,13 @@
...
 
-# These users are the default owners for everything in the repo.
-# They will be requested for review when someone opens a pull request.
+# These users are the default owners for everything in the repo
+# They will be requested for review when someone opens a pull request
 
-# You can also specify code owners for specific directories or files.
-# For example:
+* @zksync-sdk/devxp
+/content
```
Comment (MexicanAce) [inline .github/CODEOWNERS:7]: Do these specific directories need to have the GitHub Team specified? Or is it okay to leave them blank like that?

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -1,9 +1,13 @@
...
+# This CODEOWNERS file sets the individuals responsible for code in the sdk-docs repository
 
-# These users are the default owners for everything in the repo.
-# They will be requested for review when someone opens a pull request.
+# These users are the default owners for everything in the repo
+# They will be requested for review when someone opens a pull request
 
-# You can also specify code owners for specific directories or files.
-# For example:
+* @zksync-sdk/devxp
```
Comment (MexicanAce) [inline .github/CODEOWNERS:6]: Should we expand this to also include `txfusion`?
```suggestion
* @zksync-sdk/devxp @zksync-sdk/txfusion 
```

Comment (itsacoyote) [reply .github/CODEOWNERS:6]: Wasn't sure if they owned all of the docs as well to require them as part of the reviewers for `*`.

File: .github/CODEOWNERS (hunk 1 above)
Comment (itsacoyote) [reply .github/CODEOWNERS:7]: The ruleset here is that devxp are owners for "everything" except for `/content` and `/public`. This way we ensure we review changes to the project structure but aren't required on the content and public folders so we don't hold up reviews in there. 

If txFusion should be owners for all the content in the /content directory I can also specify that.

File: .github/CODEOWNERS (hunk 2 above)
Comment (MexicanAce) [reply .github/CODEOWNERS:6]: Ah, based on the other comment, limiting docs/repo structure changes can go through `devxp` for now

File: .github/CODEOWNERS (hunk 1 above)
Comment (MexicanAce) [reply .github/CODEOWNERS:7]: Agreed about the split, and also think it's a good idea for both `devxp` and `txfusion` to be listed as reviewers for the `/content` directory.

Can we also add comments to the `CODEOWNERS` file with your explanation above?

**Review by MexicanAce — APPROVED** [review]
LGTM

**Pull request merged**